### PR TITLE
Add toolchain support to release notes

### PR DESCRIPTION
**Pull request by bmuskalla**

### CODE REVIEW

**Comment by bmuskalla** [comment]
@bot-gradle test QF

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -39,6 +39,37 @@ For Java, Groovy, Kotlin and Android compatibility, see the [full compatibility
...
 
+## Toolchains for JVM projects
+
+By default, Gradle uses the same Java version for running Gradle itself and building JVM projects. 
+
+This is not always desirable. Building projects with different Java versions on different developer machines and CI servers may lead to unexpected issues Additionally, you may want to build a project using a Java version that running Gradle is not compatible with.
+
+Before this release, it required [several steps](userguide/building_java_projects.html#compiling_and_testing_java_67) to configure a different Java version for compilation, testing, generating Javadoc, and executing applications.  
+
+This release introduces a new concept called Java Toolchains to Gradle. A Java Toolchain (from now on referred to simply as toolchain) is a set of tools (e.g. a java compiler), taken from a local JRE/JDK installation and used to configure the build.
```
Comment (pioterj) [inline subprojects/docs/src/docs/release/notes.md:50]: "java compiler" -> "Java compiler"

File: subprojects/docs/src/docs/release/notes.md [hunk 2]
```diff
@@ -39,6 +39,37 @@ For Java, Groovy, Kotlin and Android compatibility, see the [full compatibility
...
+
+But instead of manually specifying this for the various tasks, toolchains provide a holistic set of requirements that define what the build needs.
```
Comment (pioterj) [inline subprojects/docs/src/docs/release/notes.md:52]: Remove "But"

File: subprojects/docs/src/docs/release/notes.md [hunk 3]
```diff
@@ -39,6 +39,37 @@ For Java, Groovy, Kotlin and Android compatibility, see the [full compatibility
...
+
+```
+java {
+    toolchain {
+        languageVersion = JavaLanguageVersion.of(14)
+    }
+}
+```
+
+Compile tasks use `javac` as their compiler, test and exec tasks may use the `java` command while `javadoc` will be used to generate documentation. All taken from the selected toolchain above.
```
Comment (pioterj) [inline subprojects/docs/src/docs/release/notes.md:64]: Let's make the first sentence consistent so it does not mix "use", "may use" and "will be used" for equivalent items.

File: subprojects/docs/src/docs/release/notes.md [hunk 4]
```diff
@@ -39,6 +39,37 @@ For Java, Groovy, Kotlin and Android compatibility, see the [full compatibility
...
+
+With this, Gradle handles setting up the build and environment accordingly. It will detect locally installed Java versions to run the build according to the specified requirements. By default, Gradle supports common installation locations on various operating systems as well as popular package managers like asdf-vm, jabba and SDKMAN!. If no matching Java version is available locally, it will download a matching JDK from AdoptOpenJDK.
```
Comment (pioterj) [inline subprojects/docs/src/docs/release/notes.md:68]: "With this" -> "Using the snippet above"

**Review by pioterj — CHANGES_REQUESTED** [review]

**Comment by ljacomet** [comment]
@pioterj I addressed your comments and also moved the demo to be right before the documentation links, after all the text. I could not find a good place for it earlier without breaking the flow of the explanation.

Let me know what you think of the content now.

**Comment by pioterj** [comment]
@ljacomet Could you give me a link to the rendered version?

**Comment by ljacomet** [comment]
Changes included in #14489

**Pull request closed without merging**